tools/SetupFlow/DevHome.SetupFlow/Controls/SetupShell.xaml

```diff
@@ -15,93 +15,82 @@
         <converters:BoolToObjectConverter x:Key="StepperBoolToGlyphBrushConverter" TrueValue="{ThemeResource AccentFillColorDisabledBrush}" FalseValue="{ThemeResource AccentFillColorDefaultBrush}" />
         <converters:BoolToVisibilityConverter x:Key="NegatedBoolToVisibilityConverter" TrueValue="Collapsed" FalseValue="Visible" />
     </UserControl.Resources>
-    <Grid RowSpacing="6" x:Name="ShellContent">
+    <Grid x:Name="ShellContent">
         <Grid.RowDefinitions>
-            <RowDefinition Height="auto"/>
-            <RowDefinition Height="auto"/>
             <RowDefinition Height="auto"/>
             <RowDefinition Height="*"/>
         </Grid.RowDefinitions>
-        <!-- Title -->
-        <StackPanel Grid.Row="0">
+
+        <StackPanel Spacing="6" Visibility="{x:Bind HeaderVisibility, Mode=OneWay}">
+            <!-- Title -->
             <TextBlock
                 Style="{ThemeResource SubtitleTextBlockStyle}"
                 TextWrapping="WrapWholeWords"
-                Visibility="{x:Bind UseOrchestratorTitle, Mode=OneWay}"
-                Text="{x:Bind Orchestrator.FlowTitle}">
+                Text="{x:Bind Title, Mode=OneWay}">
                 <i:Interaction.Behaviors>
                     <behaviors:TextBlockAutomationBehavior RaiseLiveRegionChangedEvent="True" LiveSetting="Assertive" />
                 </i:Interaction.Behaviors>
             </TextBlock>
+            <Grid>
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition Width="auto"/>
+                    <ColumnDefinition Width="*"/>
+                </Grid.ColumnDefinitions>
+                <!--
+                    TODO: https://github.com/microsoft/devhome/issues/1621
+                    Promote stepper to a separate control
+                -->
+                <!-- Stepper showing the progress within the flow -->
+                <StackPanel Margin="0, 20, 0, 0" Visibility="{x:Bind Orchestrator.CurrentPageViewModel.IsStepPage, Mode=OneWay}">
+                    <ItemsRepeater ItemsSource="{x:Bind Orchestrator.SetupStepPages, Mode=OneWay}">
+                        <ItemsRepeater.Layout>
+                            <StackLayout Orientation="Horizontal" Spacing="5"/>
+                        </ItemsRepeater.Layout>
+                        <ItemsRepeater.ItemTemplate>
+                            <DataTemplate x:DataType="viewModels:SetupPageViewModelBase">
+                                <StackPanel Spacing="8" Orientation="Horizontal"

```

Comment:
I actually don't think we need a new line for every xaml property (so much wasted space if they all fit on one line! :smile:) but I do think the spacing on this element needs to be fixed.

Reply:
Yeah.  The setup tool has a mix of both formats.  I'd vote for all going one line per attribute.